internal/core/software_renderer.rs

```diff
@@ -145,6 +150,32 @@ impl SoftwareRenderer {
         }
     }
 
+    /// Create a new Renderer for a given window.
+    ///
+    /// The `repaint_buffer_type` parameter specify what kind of buffer are passed to [`Self::render`]
+    ///
+    /// The `window` parameter can be coming from [`Rc::new_cyclic()`](alloc::rc::Rc::new_cyclic())
+    /// since the `WindowAdapter` most likely own the Renderer
+    #[doc(hidden)]
+    pub fn new_without_window(repaint_buffer_type: RepaintBufferType) -> Self {
+        Self {
+            window: RefCell::new(None),
+            repaint_buffer_type,
+            partial_cache: Default::default(),
+            force_dirty: Default::default(),
+            force_screen_refresh: Default::default(),
+            prev_frame_dirty: Default::default(),
+        }
+    }
+
+    /// Sets the window to be use for future rendering operations. Call this before calling
+    /// rendering.
+    #[doc(hidden)]
+    pub fn set_window(&self, window: &crate::api::Window) {
```

Comment:
```suggestion
    pub(crate) fn set_window(&self, window: &crate::api::Window) {
```

Reply:
The software renderer in the winit backend needs to call this, too :-(.

Another option would be to create a "new" SoftwareRenderer type with a "v2" type of API, not export it in the public Slint Rust API, but export it as part of the core crate private API. Then we implement the old public API top of the new API.

Reply:
Other options to make sure that noone calls this:
```suggestion
    pub fn set_window(&self, window: &crate::api::Window, _: crate::InternalToken) {
```

where  InternalToken is
```rust
/// This struct can only be named if one import directly `i-slint-corelib` and function that takes it 
/// as a parameter cannot be called directly without depending on the internal library.
struct InternalToken {}
``` 

I *think* such tricks should work, but i'm not 100% sure

Reply:
It is still possible to call with `unsafe`.

Another way to do it is not to hve it as a member function, but as a module-level function that is not re-exported

Reply:
The module-level function is the approach I went with for now.

Reply:
Ok, the module-level approach is rather ugly but not any "safer" in terms of risk of somebody calling this.

> Just asking because i think it might be worth to make sure nobody calls this (or you think just having it as hidden is fine?)

I'll stick with doc(hidden). Whether it's the module-level or struct-level function - it's the same "risk".